x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx

```diff
@@ -67,11 +69,21 @@ export function ApmServiceContextProvider({
     end,
   });
 
+  const preferred = usePreferredDataSourceAndBucketSize({
```

Comment:
We talked about splitting this into `usePreferredDataSource` and `usePreferredBucketSize`. Is that still the plan for the follow up?

Reply:
Hmm, I see. In the case of single value metrics (eg average latency in the transactions table) we don't want `numBuckets` to be 20 or 100 which are the only possible values right now afaict. See here for instance:

https://github.com/elastic/kibana/blob/cdeeff28026b370fab9aafce97071b263a14d17b/x-pack/plugins/apm/public/components/shared/transactions_table/index.tsx#L146-L152

I suggest changing it to `numBuckets: 1`.